Contact Diary - Onboarding screen (EXPOSUREAPP-4154)

axelherbstreith opened the pull request:
Added basic functionalities for the contact diary onboarding screen

axelherbstreith committed:
Added basic layout

axelherbstreith committed:
Added translation support

axelherbstreith committed:
Added darkmode images

axelherbstreith committed:
Included dimens in layout

axelherbstreith committed:
Added navigation to onboarding fragment

axelherbstreith committed:
Changed AndroidManifest to support new contact diary activity

axelherbstreith committed:
Updated translation files

axelherbstreith committed:
Added missing image
Updated text styling
Changed layout naming

axelherbstreith committed:
Added back navigation

axelherbstreith committed:
Updated layout

axelherbstreith committed:
Added direct link to privacy information

d4rken commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/information/InformationPrivacyFragment.kt:
```diff
@@ -27,7 +28,14 @@ class InformationPrivacyFragment : Fragment(R.layout.fragment_information_privac
         binding.informationPrivacyHeader.headerButtonBack.buttonIcon.setOnClickListener {
-            (activity as MainActivity).goBack()
+
+            if (activity is (MainActivity)) {
```
Can't we just `popBackstack()` here? It's not so nice that the fragment needs to know about the different activities.

d4rken commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,205 @@
+        android:layout_height="match_parent">
+
+
+        <include
```
Please evaluate using a normal Toolbar instead of the layout include.

BMItr committed:
repair PR commit

BMItter commented:
Tested on Pixel 2 Device.
Looking good so far.

axelherbstreith committed:
Changed back navigation

BMItr committed:
repair

Oliver-Zimmerman commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,205 @@
+                android:layout_height="wrap_content">
+
+
+                <ImageView
```
In Figma it looks like this image needs to touch both sides of the screen, rather than be between guidelines.

axelherbstreith replied:
Image is constraint to both sides, depending on the screen ratio of the test device there can be white margins

chiljamgossow replied:
There should not be white margins, please check

axelherbstreith replied:
Its for all images within the app. Not just for the one in the onboarding screen. Depends on the screen ratio. May need to consider changing the constraints for all images then.

SamuraiKek commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/onboarding/ContactDiaryOnboardingFragmentModul.kt:
```diff
@@ -0,0 +1,23 @@
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelKey
+
+@Module
+abstract class ContactDiaryOnboardingFragmentModul {
```
Typo in name. Should be Module*

SamuraiKek commented:
Lgtm, except for the small typo. The image is touching the sides of the screen as in the design.

Oliver-Zimmerman commented:
@SamuraiKek @axelherbstreith perhaps it is device dependant. On my Pixel 5 there is a margin on both ends.

axelherbstreith committed:
Changed information text

axelherbstreith committed:
Fixed typo in module

axelherbstreith committed:
updated strings

axelherbstreith committed:
Aligned back navigation

Oliver-Zimmerman reviewed (approved):
LGTM 👍

chiljamgossow commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/ContactDiaryActivity.kt:
```diff
@@ -43,4 +43,11 @@ class ContactDiaryActivity : AppCompatActivity(), HasAndroidInjector {
+    /**
+     * Function is called from back buttons in fragments.
+     */
+    fun goBack() {
```
What do we need this for?

axelherbstreith replied:
is removed now

chiljamgossow commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,232 @@
+        <ScrollView
+            android:layout_width="@dimen/match_constraint"
+            android:layout_height="@dimen/match_constraint"
+            android:layout_marginBottom="15dp"
```
Dimens should be even. Please a constant from the dimens file, in this case spacing_small

chiljamgossow commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,232 @@
+                    android:focusable="true"
+                    android:src="@drawable/ic_contact_diary_illustration_onboarding"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.573"
```
What is the bias for?

chiljamgossow commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,232 @@
+                    layout="@layout/include_contact_diary_onboarding_row"
+                    android:layout_width="@dimen/match_constraint"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="24dp"
```
Please use values from dimens

chiljamgossow commented on Corona-Warn-App/src/main/res/layout/include_contact_diary_onboarding_row.xml:
```diff
@@ -0,0 +1,49 @@
+            android:id="@+id/contact_diary_row_icon"
+            android:layout_width="@dimen/icon_size_risk_card"
+            android:layout_height="@dimen/icon_size_risk_card"
+            android:layout_marginTop="4dp"
```
Please use dimens

axelherbstreith committed:
Changed margins
Remove unused back navigation functionality

SamuraiKek commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,228 @@
+                    android:layout_width="@dimen/match_constraint"
+                    android:layout_height="1dp"
+                    android:layout_marginTop="@dimen/spacing_small"
+                    android:layout_marginBottom="20dp"
```
Missed a spacing here.

SamuraiKek commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,228 @@
+
+
+        <Button
+            android:id="@+id/contact_diary_onboarding_next_button"
```
The spacing for the button should also use dimensions instead of hardcoded values.

axelherbstreith committed:
Added missing dimens

ralfgehrer commented:
<img width="396" alt="Screenshot 2020-12-15 at 11 59 19" src="https://user-images.githubusercontent.com/56824/102206469-f9cacc80-3ecc-11eb-94dc-0e19405a892d.png">
Image on top should be borderless according to figma

ralfgehrer commented:
<img width="398" alt="Screenshot 2020-12-15 at 12 01 04" src="https://user-images.githubusercontent.com/56824/102206621-3696c380-3ecd-11eb-9451-20676b45f953.png">

Padding on top of and below of the privacy consent is different and should be aligned

kolyaopahle commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/onboarding/ContactDiaryOnboardingFragment.kt:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    override fun onResume() {
+        super.onResume()
```
this does nothing?

ralfgehrer replied:
please remove

kolyaopahle commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/onboarding/ContactDiaryOnboardingFragmentModule.kt:
```diff
@@ -0,0 +1,23 @@
+    @Binds
+    @IntoMap
+    @CWAViewModelKey(ContactDiaryOnboardingFragmentViewModel::class)
+    abstract fun contactDiaryOnboardingFragmentVM(
```
why are both defined?

kolyaopahle commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,228 @@
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+</layout>
```
\n

kolyaopahle commented on Corona-Warn-App/src/main/res/layout/include_contact_diary_onboarding_row.xml:
```diff
@@ -0,0 +1,49 @@
+            android:layout_marginStart="@dimen/spacing_small"
+            android:text="@{body}"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.0"
```
probably unnecessary

kolyaopahle commented on Corona-Warn-App/src/main/res/layout/include_contact_diary_onboarding_row.xml:
```diff
@@ -0,0 +1,49 @@
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+</layout>
```
\n

kolyaopahle commented on Corona-Warn-App/src/main/res/layout/include_contact_diary_privacy_card.xml:
```diff
@@ -0,0 +1,65 @@
+            app:layout_constraintTop_toBottomOf="@+id/contact_diary_privacy_card_second_section_title" />
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
\n

kolyaopahle commented on Corona-Warn-App/src/main/res/layout/include_contact_diary_privacy_card.xml:
```diff
@@ -0,0 +1,65 @@
+            style="@style/subtitle"
+            android:layout_width="@dimen/match_constraint"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="24dp"
```
use dimens

kolyaopahle commented on Corona-Warn-App/src/main/res/values-de/strings.xml:
```diff
@@ -1496,4 +1496,45 @@
     <string name="contact_diary_homescreen_card_button_text">Tagebuch Führen</string>
 
+    <!-- XBUT: Button text for the contact diary card button in the homescreen   -->
+    <string name="contact_diary_onboarding_button_text">Tagebuch Führen</string>
```
maybe move these to the diary strings file

ralfgehrer replied:
Please do, otherwise it will be a big mess

SamuraiKek replied:
These strings were added by me in the homescreen diary card ticket. I thought of putting them here because I thought since the card is part of the homescreen and not the diary stuff, it's probably better to have them in the main strings files. 

Anyway, I can see @axelherbstreith has moved them to the diary strings file already, maybe just a small observation: I think these strings should be completely removed from here and put in the (de) contact_diary_strings file. Sorry for giving you extra work!

axelherbstreith replied:
Done :)

kolyaopahle commented on Corona-Warn-App/src/main/res/values/strings.xml:
```diff
@@ -1489,11 +1489,49 @@
     <string name="contact_diary_homescreen_card_button_text" />
 
+    <!-- XBUT: Button text for the contact diary card button in the homescreen   -->
+    <string name="contact_diary_onboarding_button_text"/>
```
maybe move these to the diary strings file

axelherbstreith committed:
Added missing dimens
Added updated icons
Added final layout
Updated strings
Updated viewmodel
removed unused code

BMItr committed:
Merge branch 'feature/4152-contact-diary' into feature/4154-contact-diary-onboarding-screen

axelherbstreith committed:
Moved contact diary strings

kolyaopahle commented on Corona-Warn-App/src/main/res/layout/contact_diary_onboarding_fragment.xml:
```diff
@@ -0,0 +1,221 @@
+                <View
+                    android:id="@+id/contact_diary_onboarding_second_divider"
+                    android:layout_width="@dimen/match_constraint"
+                    android:layout_height="1dp"
```
do we have any dimens setup for spacers like this? otherwise you could create one

axelherbstreith committed:
Moved strings
Added line breaks
added dimens to dividers
